app/pages/resource/reservation-form/ReservationFormContainer.js

```diff
@@ -0,0 +1,17 @@
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import ReservationForm from './ReservationForm';
+
+const resourceSelector = (state, props) => props.resource;
```

Comment:
> The main reason is that each file should have one clear purpose and not do multiple things.

The definition of "one clear purpose" is quite vague. "Select the resource from the props" is one clear purpose, so the line `const resourceSelector = (state, props) => props.resource;` could be in its own file. Then again, "Render the reservation form for a given resource" is also one clear purpose, so the selector, actions, and rendering could be all in one file.

I think dividing the code to different files has two pros:
1. Code sharing. It's difficult to for example reuse a part of the form rendering code or a selector if all of these are in the same file.
2. Code readability.

I don't think selectors, actions, etc. as needed by the containers are typically reusable. And if they are, they should probably be located in a place that is more logical for reusable selectors/actions/whatever, not in 'resourceFormSelectors.js'.

So then the main advantage is code readability. If the selectors are small, moving them to another file doesn't improve readability. If the rendering code is small, moving it to another file also doesn't improve readability. Moving them to separate files only improves readability if those parts become larger / more complicated. On that point we could by default have the selectors, actions, and rendering all in one file and then once one of those parts becomes complicated, move it to its own file. For example in this file's case the rendering is quite complex so it should be in its own file as it currently is.

---

I recall the Redux or React-Redux docs presenting a different, quite interesting approach at some point. The idea behind it was that connecting a component to the state creates a component with new functionality, therefore the new component should be named accordingly. So instead of having a ReservationForm and a ReservationFormContainer we could call them ReservationForm and ResourceReservationForm (as I've understood the container just populates the form with a resource's data?).

Or maybe a slightly different approach with the following components (I'm not super familiar with the reservation form functionality so some assumptions I've made could be off):

- ReservationForm: The current ReservationForm.
- ReservationEditForm: Creates a form populated with the given reservation's data and on submit dispatches an edit reservation action.
- ReservationCreateForm: Creates an empty form and on submit dispatches a create reservation action.
- ResourceReservationCreateForm: Creates a ReservationCreateForm populated with a given resource's data. These two ...CreateForm components could be just one component if we don't need to render non-populated ReservationCreateForms anywhere.

The biggest challenge I've noticed with this approach is that often it's difficult to come up with a name other than "SomethingContainer" for the connected component. But maybe that's a symptom of us coupling the containers and components too tightly?